docs/proposals/resize-when-creating-by-group.md

```diff
@@ -0,0 +1,127 @@
+
+- If creating a PVC has a `resize.topolvm.io/pre-resize-group-by` annotation and the label specified in the annotation exists, existing PVCs with matching values are determined to be in the same group.
+- If no PVCs are determined to be in the same group, or if the capacity of creating PVC is larger than existing any PVC in the same group, the capacity of the original PVC is used as is.
+
+Examples are given below.
+
+```yaml
+### existing PVCs (excerpted)
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-1
+  labels:
+    label-hoge: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 20Gi
+
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-2
+  labels:
+    label-hoge: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 16Gi
+
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-y-1
+  labels:
+    label-hoge: group-y
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 30Gi
+```
+
+When creating the following new PVC, `pvc-x-1` and `pvc-x-2` with `label-hoge: group-x` are determined to be in the same group, and `pvc-y-1` is determined to be in a different group. Therefore, the PVC is created with **20Gi** based on `pvc-x-1`, which has the largest capacity.
+
+```yaml
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-3
+  labels:
+    label-hoge: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 10Gi
+```
+
+When creating the following new PVC, `pvc-y-1` with `label-hoge: group-y` is a candidate, but since the new PVC's size(50Gi) is larger than the existing(30Gi), the PVC will be created with **50Gi**.
+
+```yaml
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-y-2
+  labels:
+    label-hoge: group-y
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 50Gi
+```
+
+### How to use
+
+- Add `resize.topolvm.io/pre-resize-group-by` annotation to the PVC template to which you want to apply the feature.
+- In most cases, annotations and labels are not added to PVCs that have already been created based on a PVC template. In such cases, it is necessary to manually add `resize.topolvm.io/pre-resize-group-by` annotation to existing PVCs in addition to the PVC template.
+
+## Design details
+
+Implementation method.
+
+### Option A) use mutating webhook for PVC
+
+Pros:
+- PVCs are created with the required capacity specified from the beginning, even a restore program that checks the free capacity of the volume will not cause problems.
+- PVCs are created with the required capacity specified from the beginning, so they will not become unscalable later due to insufficient capacity when the capacity per node is had, as is the case of TopoLVM.
```

Comment:
Sorry, but I could not understand this line.
Especially, what do you mean by saying "when the capacity per node is had"?

Reply:
How about changing the line as follows?
```suggestion
- We do not need to worry about expansion failure after scheduling PVs because new PVC is created with the sufficient capacity.
```